crates/dbs-utils/src/rate_limiter.rs

```diff
@@ -618,6 +618,7 @@ mod tests {
         thread::sleep(Duration::from_millis(123));
         assert_eq!(tb.reduce(1), BucketReduction::Success);
         assert_eq!(tb.reduce(100), BucketReduction::Success);
+        thread::sleep(Duration::from_millis(10));
```

Comment:
sleep（）means there are fundamental issues behind, maybe we should use some synchronization mechanism.

Reply:
The `cargo test` is currently running with thread number 1 (--test-threads 1). This is to solve the problem that the should_panic test case fails. 

The failure here is because the attempt to consume tokens from insufficient buckets should fail, but here is enough. Tokens are automatically replenished over time. 

the total of tokens: 1000 
1st, reduce 123  -> Success,  now tokens are 877 . #614.
second, reduce 1000 -> Failure, not enough 1000.  #616.

sleep 123,  #618.

third, reduce 1 -> Success, now tokens are 876. #619.
fourth, reduce 100 -> Success, now tokens are 776.  #620.
5th, reduce 1000 ->  auto_replenish,  should be failed, but returned Success.

So too much is replenished here